### PR TITLE
Icu 10926 add dynamic credential fields in sb model

### PR DESCRIPTION
**Pull request by DhariniJeeva**
:tickets: [Jira ticket](https://hashicorp.atlassian.net/browse/ICU-10926)

## Description

This PR adds support for dynamic credentials in storage bucket creation.

## Screenshots (if appropriate):
Screenshot 2023-09-08 at 2 45 37 PM

 
 
## Checklist:

- [ ] ~I have added before and after screenshots for UI changes~
- [x] I have added JSON response output for API changes
- [ ] ~I have added steps to reproduce and test for bug fixes in the description~
- [ ] ~I have commented on my code, particularly in hard-to-understand areas~
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: addons/api/addon/generated/models/storage-bucket.js [hunk 1]
```diff
@@ -79,6 +79,13 @@ export default class GeneratedStorageBucketModel extends BaseModel {
   })
   secrets_hmac;
 
+  @attr('boolean', {
+    isNestedAttribute: true,
+    description: '',
```
Comment (cameronperera) [inline addons/api/addon/generated/models/storage-bucket.js:84]: Do we have a description we can provide here?

**Review by cameronperera — APPROVED** [review]
LGTM. Left a non-blocking comment.

File: addons/api/tests/unit/serializers/storage-bucket-test.js [hunk 2]
```diff
@@ -57,6 +65,44 @@ module('Unit | Serializer | storage bucket', function (hooks) {
...
+      worker_filter: 'workerfilter',
+      region: 'eu-west-1',
+      access_key_id: '',
+      secret_access_key: '',
+      secrets_hmac: 'hmac',
+      disable_credential_rotation: true,
+      role_arn: 'arn',
+      role_external_id: null,
+      role_session_name: null,
+      role_tags: [],
```
Comment (ZedLi) [inline addons/api/tests/unit/serializers/storage-bucket-test.js:86]: Can we have these fields be non null so we can show that they get serialized correctly?

**Review by ZedLi — APPROVED** [review]
Looks good!

File: addons/api/tests/unit/serializers/storage-bucket-test.js (hunk 2 above)
Comment (calcaide) [reply addons/api/tests/unit/serializers/storage-bucket-test.js:86]: Agreed with Zed's comment. We should make sure we serialize correctly those files with and without values.

**Review by calcaide — APPROVED** [review]
LGTM!! great work! left a none blocking comment

File: addons/api/tests/unit/serializers/storage-bucket-test.js (hunk 2 above)
Comment (DhariniJeeva) [reply addons/api/tests/unit/serializers/storage-bucket-test.js]: updated the attr type to use `object-as-array` transform as that is the expected backend format from the request.

**Pull request merged**